test/test_stepsize.py

```diff
@@ -185,7 +185,7 @@ def test_overshoot_milestone(nb_its=1, target=4):
         value=current_value, nb_iterations=nb_its, t=0.9
     )
 
-    expected_value = 1.3
+    expected_value = 1.3 - current_value
```

Comment:
For this test, you don't even need a growth factor > 1. Try this:

growth factor = 1

curent_dt_value = 100000  very large

current_time = 1  # next timestep will overshoot the milestone of 1.3

expected_value = 1.3 - current_time # you need to fix this !!!